extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/JdbcBasePersistenceImpl.java

```diff
@@ -189,6 +185,18 @@ private void persistEntity(
     }
   }
 
+  private int execute(Object executor, String query) throws SQLException {
+    if (executor instanceof Statement) {
+      // used for running in transaction
+      return ((Statement) executor).executeUpdate(query);
+    } else if (executor instanceof DatasourceOperations) {
+      // used for atomic queries
```

Comment:
The term "atomic" is misleading in this context. It usually means "all or nothing" but in that sense it applies to any PostgreSQL DML committed via JDBC.